### PR TITLE
More misc libgit2 tests

### PR DESCRIPTION
**Pull request by kshyatt**
Let's get this folder over 70% covered! Tests for OIDs and blobs and tree entries.

### CODE REVIEW

File: test/libgit2.jl [hunk 1]
```diff
@@ -294,7 +294,12 @@ mktempdir() do dir
                 try
                     @test commit_oid1 == LibGit2.GitHash(cmt)
                     short_oid1 = LibGit2.GitShortHash(string(commit_oid1))
-                    @test cmp(commit_oid1,short_oid1) == 0
+                    @test hex(commit_oid1) == hex(short_oid1)
+                    @test cmp(commit_oid1, short_oid1) == 0
+                    @test cmp(short_oid1, commit_oid1) == 0
+                    @test !(short_oid1 < commit_oid1)
```
Comment (ararslan) [inline test/libgit2.jl]: `@test short_oid1 >= commit_oid1`?

Comment (kshyatt) [inline test/libgit2.jl]: I wanted to make sure `isless` gets called.

Comment (ararslan) [inline test/libgit2.jl]: You could call `isless`

Comment (kshyatt) [inline test/libgit2.jl]: Yes but I don't wanna

Comment (ararslan) [inline test/libgit2.jl]: Okay

**Comment by kshyatt** [comment]
Gist of failed build here https://gist.github.com/kshyatt/354142aa4118dbdd29458910aea42b2e

**Review by ararslan — APPROVED** [review]
🐃

**Pull request merged**